datafusion/sql/src/planner.rs

```diff
@@ -502,7 +502,14 @@ pub fn object_name_to_table_reference(
     let ObjectName(idents) = object_name;
     idents_to_table_reference(idents, enable_normalization)
 }
-
+fn substitute_tilde(cur: String) -> String {
```

Comment:
I am a little worried about the securoty implications of putting tilde expansion into the sql planner itself as this will cause *all* identifiers that tsart with `~` to be replaced

like `SELECT * from ~foo`

I think this tilde expansion should be done only in the datafusion-cli code -- specifically I think somewhere like https://github.com/apache/arrow-datafusion/blob/e62240969135e2236d100c8c0c01546a87950a80/datafusion-cli/src/catalog.rs#L163 (which is where it handles converting unknown table names to urls)

Reply:
Got it, I will change it now